gcp-observability/src/main/java/io/grpc/gcp/observability/GcpObservability.java

```diff
@@ -73,17 +116,104 @@ public void close() {
       if (instance == null) {
         throw new IllegalStateException("GcpObservability already closed!");
       }
+      unRegisterStackDriverExporter();
       LoggingChannelProvider.shutdown();
       LoggingServerProvider.shutdown();
       sink.close();
       instance = null;
     }
   }
 
-  private GcpObservability(Sink sink,
+  private void setProducer(
+      InternalLoggingChannelInterceptor.Factory channelInterceptorFactory,
+      InternalLoggingServerInterceptor.Factory serverInterceptorFactory) {
+    if (config.isEnableCloudLogging()) {
+      clientInterceptors.add(channelInterceptorFactory.create());
+      serverInterceptors.add(serverInterceptorFactory.create());
+    }
+    if (config.isEnableCloudMonitoring()) {
+      clientInterceptors.add(
+          InternalCensusStatsAccessor.getClientInterceptor(true, true, true, true));
+      tracerFactories.add(
+          InternalCensusStatsAccessor.getServerStreamTracerFactory(true, true, true));
+    }
+    if (config.isEnableCloudTracing()) {
+      clientInterceptors.add(InternalCensusTracingAccessor.getClientInterceptor());
+      tracerFactories.add(InternalCensusTracingAccessor.getServerStreamTracerFactory());
+    }
+
+    if (!clientInterceptors.isEmpty()
```

Comment:
Based on the latest change in the other PR, you should remove this check in the `if`. Even if empty you should now set the interceptors or tracers in the GlobalInterceptors

Reply:
Added unit test to verify empty lists are returned when `grpcInit` is called and none of the observability features are enabled.